spotinst/resource_spotinst_ocean_spark.go

```diff
@@ -213,9 +225,60 @@ func deleteSparkCluster(ctx context.Context, resourceData *schema.ResourceData,
 		log.Printf("===> Cluster delete configuration: %s", json)
 	}
 
-	if _, err := meta.(*Client).ocean.Spark().DeleteCluster(ctx, input); err != nil {
+	if _, err := oceanSparkClient.DeleteCluster(ctx, input); err != nil {
 		return fmt.Errorf("[ERROR] onDelete() -> Failed to delete cluster: %s", err)
 	}
 
+	if forceDelete {
+		log.Printf("===> Cluster has been force deleted: %s <===", resourceData.Id())
+		return nil
+	}
+
+	if err := waitUntilClusterDeleted(ctx, resourceData, oceanSparkClient); err != nil {
+		return err
+	}
+
 	return nil
 }
+
+func waitUntilClusterDeleted(ctx context.Context, resourceData *schema.ResourceData, oceanSparkClient spark.Service) error {
+	timeout := time.After(deleteTimeout)
+	for {
+		select {
+		case <-timeout:
+			return fmt.Errorf("timed out waiting for cluster deletion")
+		default:
+			isDeleted, err := isClusterDeleted(ctx, resourceData, oceanSparkClient)
+			if err != nil {
+				return fmt.Errorf("could not verify cluster deletion, %w", err)
+			}
+
+			if isDeleted {
+				return nil
+			}
+
+			time.Sleep(sleepBetweenDeleteChecks)
```

Comment:
Instead of doing a `time.Sleep` a slightly more common approach is to utilize a ticker.

One of the major benefits of a ticker is that if the `isClusterDeleted` block runs for longer than the sleep period the ticker will adjust and skip the next invocation.

```golang
	timeout := time.After(deleteTimeout)
	checkDeleted := time.NewTicker(sleepBetweenDeleteChecks)
	defer checkDeleted.Stop()

	for {
		select {
		case <-timeout:
			return fmt.Errorf("timed out waiting for cluster deletion")
		case <-checkDeleted.C:
			isDeleted, err := isClusterDeleted(ctx, resourceData, oceanSparkClient)
			if err != nil {
				return fmt.Errorf("could not verify cluster deletion, %w", err)
			}

			if isDeleted {
				return nil
			}
		}
	}
```